File: final_project_server/models/item.ts
import mongoose from "mongoose";

const itemSchema = new mongoose.Schema({
  name: {
    type: String,
    trim: true,
    required: true,
  },
  description: {
    type: String,
    trim: true,
    required: true,
  },
  rarity: {
    type: String,
    required: true,
  },
  cost: {
    type: Number,
    required: true,
  },
  sprite: {
    type: String,
    required: true,
  },
});
itemSchema.set("toJSON", {
  transform: (_document, returnedObject) => {
    //@ts-ignore
    returnedObject.id = returnedObject._id.toString();
    //@ts-ignore
    delete returnedObject._id;
    //@ts-ignore
    delete returnedObject.__v;
  },
});

module.exports = mongoose.model("Item", itemSchema);

File: final_project/lib/src/models/item.dart
import 'package:json_annotation/json_annotation.dart';

part 'item.g.dart';

@JsonSerializable()
class Item {
  Item(
    this.name,
    this.description,
    this.rarity,
    this.cost,
    this.sprite,
    this.id,
  );

  String name;
  String description;
  String rarity;
  int cost;
  String sprite;
  String id;

  factory Item.fromJson(Map<String, dynamic> json) => _$ItemFromJson(json);

  Map<String, dynamic> toJson() => _$ItemToJson(this);
}
